### PR TITLE
World api fixes

### PR DESCRIPTION
**Pull request by jaimedp**

### CODE REVIEW

File: src/service/world-api-adapter.js [hunk 1]
```diff
@@ -249,6 +250,34 @@ module.exports = function (config) {
...
+        * Updates a user from a given world (only one user at a time)
+        *
+        * Supported formats:
+        * ws.updateUser({ userId: 'b1c19dda-2d2e-4777-ad5d-3929f17e86d3', role: 'leader' });
+        *
+        * @param user {object} user object with userId and the new role
+        * @param options {object} (Optional) Options object to override global options
+        *
+        */
+        updateUser: function (user, options) {
```
Comment (narenranjit) [inline src/service/world-api-adapter.js]: Can you update anything besides the role?

Comment (jaimedp) [inline src/service/world-api-adapter.js]: I think team, but it's in development

**Comment by mmrj** [comment]
i think it's just role

**Comment by narenranjit** [comment]
@jaimedp We can't merge this yet right? Because of game-world url change?

File: src/service/world-api-adapter.js [hunk 2]
```diff
@@ -223,6 +219,24 @@ module.exports = function (config) {
...
+        *
+        */
+        load: function (worldId) {
+            if (!worldId || typeof worldId !== 'string') {
+                throw new Error('load needs a worldId string to load (' + worldId + ')');
+            }
+
+            serviceOptions.filter = worldId;
+
+            return this;
```
Comment (narenranjit) [inline src/service/world-api-adapter.js]: We should do a `get` and return a promise, or call this something else - the `runservice` and `dataservice` have `.load` which behaves differently

File: src/service/world-api-adapter.js [hunk 3]
```diff
@@ -507,7 +568,54 @@ module.exports = function (config) {
...
+                { url: urlConfig.getAPIPath(assignmentEndpoint) }
+            );
+
+            var params = {
+                account: opt.account,
+                project: opt.project,
+                group: opt.group
+            };
+
+            if (opt.maxUsers) {
```
Comment (narenranjit) [inline src/service/world-api-adapter.js]: Can't we use _pick(opt, [..])?

File: tests/spec/test-world-api-service.js [hunk 4]
```diff
@@ -175,17 +250,36 @@
...
                 var req = server.requests.pop();
                 req.method.toUpperCase().should.equal('DELETE');
-                req.url.should.match(/\/game\/gameid1\/users/);
+                req.url.should.match(/\/world\/gameid1\/users\/123/);
             });
         });
 
+        describe('updateUser', function () {
+            it('should patch the correct user for the correct world', function () {
+                createWorldAdapter({ filter: 'gameid1' }).updateUser({ userId:'123', role: 'abc' });
```
Comment (narenranjit) [inline tests/spec/test-world-api-service.js]: should also find and replace gameid with worldid to be consistent

**Comment by jaimedp** [comment]
This is included in assignment branch

**Pull request closed without merging**